pkg/k8s/statedb.go

```diff
@@ -142,17 +143,16 @@ func (cfg ReflectorConfig[Obj]) JobName() string {
 	return fmt.Sprintf("k8s-reflector-%s-%s", cfg.Table.Name(), cfg.Name)
 }
 
-// TransformFunc is an optional function to give to the Kubernetes reflector
+// TransformFunc is an optional function to give to the reflector
 // to transform the object returned by the ListerWatcher to the desired
 // target object. If the function returns false the object is silently
 // skipped.
 type TransformFunc[Obj any] func(statedb.ReadTxn, any) (obj Obj, ok bool)
 
-// TransformManyFunc is an optional function to give to the Kubernetes reflector
+// TransformManyFunc is an optional function to give to the reflector
 // to transform the object returned by the ListerWatcher to the desired set of
-// target objects to insert into the table. If the function returns false the object is silently
-// skipped.
-type TransformManyFunc[Obj any] func(statedb.ReadTxn, any) (objs []Obj)
+// target objects to insert or delete.
+type TransformManyFunc[Obj any] func(txn statedb.ReadTxn, deleted bool, obj any) (toInsert, toDelete iter.Seq[Obj])
```

Comment:
nit: I would appreciate some additional comments what are the assumptions for both `TransformFunc` and `TransformManyFunc`.
Is it safe to modify the object in place (for `TransformFunc`), or should you always copy it? I'm asking as in k8s starting from 1.27 it's safe to modify objects in Transform, but I see that some example tests do deep copy in our case.

Reply:
Good point. I'll expand on this. (Yes it's safe to modify since the object only exists for this function)